ui/hooks/sources.ts

```diff
@@ -33,25 +34,33 @@ export function useListSources(
           .helmRepositories;
         const buckets = (bucketsRes as ListBucketsResponse).buckets;
         const charts = (chartRes as ListHelmChartsResponse).helmCharts;
-
-        return [
-          ..._.map(repos, (r) => ({
-            ...r,
-            kind: FluxObjectKind.KindGitRepository,
-          })),
-          ..._.map(hrs, (c) => ({
-            ...c,
-            kind: FluxObjectKind.KindHelmRepository,
-          })),
-          ..._.map(buckets, (b) => ({
-            ...b,
-            kind: FluxObjectKind.KindBucket,
-          })),
-          ..._.map(charts, (ch) => ({
-            ...ch,
-            kind: FluxObjectKind.KindHelmChart,
-          })),
+        const ErrorList: ListError[] = [
+          ...repoRes.errors,
+          ...helmReleases.errors,
+          ...bucketsRes.errors,
+          ...chartRes.errors,
```

Comment:
It would be cool to include `kind` here just in case we do want to use that in the error messages.

```
const ErrorList: { [kind: string]: ListError } = {
  [FluxObjectKind.KindGitRepository]: repoRes.errors
  ...
};
```

Or merging it into the ListError somehow wdyt?

Reply:
I can merge but we will lose the type of `ListError `that we are using. will go with your suggestion.

Reply:
I have a better idea I'm thinking about merging as we merge the response will refactor that part and make you rereview